Whiskey/Functions/Resolve-WhiskeyDotNetSdkVersion.ps1

```diff
@@ -133,7 +106,277 @@ function Resolve-WhiskeyDotNetSdkVersion
         }
 
         Write-WhiskeyVerbose -Message ('[{0}] Latest LTS version resolved as: "{1}"' -f $MyInvocation.MyCommand,$resolvedVersion)
+        return $resolvedVersion
+    }
+
+    $urisToTry = @(
+        'https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/releases-index.json',
+        'https://raw.githubusercontent.com/dotnet/core/master/release-notes/releases-index.json'
+    )
+    $releasesIndex = $null
+    foreach( $uri in $urisToTry )
+    {
+        $releasesIndex =
+            Invoke-RestMethod -Uri $uri -ErrorAction Ignore |
+            Select-Object -ExpandProperty 'releases-index' -ErrorAction Ignore
+
+        if( $releasesIndex )
+        {
+            $releasesIndexUri = $uri
+        }
+    }
+
+    if( -not $releasesIndex )
+    {
+        Write-WhiskeyError -Message ('Unable to find the .NET Core releases index. We tried each of these URIs:{0} {0}* {1}{0} ' -f [Environment]::NewLine,($urisToTry -join ('{0}* ' -f [Environment]::NewLine)))
+        return
+    }
+
+    $releasesIndex =
+        $releasesIndex |
+        Where-Object { [Version]::TryParse($_.'channel-version', [ref]$null) } |
+        ForEach-Object {
+            $_.'channel-version' = [Version]$_.'channel-version'
+            $_
+        } |
+        Sort-Object -Property 'channel-version' -Descending
+
+    $Version -match '^\d+\.(?:\d+|\*)|^\*' | Out-Null
+    $matcher = $Matches[0]
+
+    $release = $releasesIndex | Where-Object { $_.'channel-version' -like $matcher } | Select-Object -First 1
+    if (-not $release -and $RollForward -eq [Whiskey.DotNetSdkRollForward]::Disable)
+    {
+        Write-WhiskeyError -Message ('.NET Core release matching "{0}" could not be found in "{1}"' -f $matcher, $releasesIndexUri)
+        return
+    }
+
+    $releasesJsonUri = $release | Select-Object -ExpandProperty 'releases.json'
+    Write-WhiskeyVerbose -Message ('[{0}] Resolving .NET Core SDK version "{1}" against known released versions at: "{2}"' -f $MyInvocation.MyCommand,$Version,$releasesJsonUri)
+
+    $releasesJson = Invoke-RestMethod -Uri $releasesJsonUri -ErrorAction Stop
+
+    $sdkVersions = & {
+        $releasesJson.releases |
+            Where-Object { $_ | Get-Member -Name 'sdk' } |
+            Select-Object -ExpandProperty 'sdk' |
+            Select-Object -ExpandProperty 'version'
+
+        $releasesJson.releases |
+            Where-Object { $_ | Get-Member -Name 'sdks' } |
+            Select-Object -ExpandProperty 'sdks' |
+            Select-Object -ExpandProperty 'version'
+    }
+
+    $desiredVersion = $null
+    $sortedSdkVersions = $null
+    if ( $Version -match '^(\d+)\.(\d+)\.(\d{1})(\d+)' )

```

Comment:
Invert this condition to `if ( $Version -notmatch '^(\d+)\.(\d+)\.(\d{1})(\d+)' )` and write an error if it's true:

```powershell
$msg = ".NET SDK version "$($Version)" is invalid. The SDK version must be in the form of a 3-part version number. See https://learn.microsoft.com/en-us/dotnet/core/versions/ for more information."
Write-WhiskeyError -Message $msg
return
```